elasticsearch/src/main/java/io/micronaut/elasticsearch/DefaultElasticsearchClientFactory.java

```diff
@@ -54,6 +63,18 @@ RestClient restClient(DefaultElasticsearchConfigurationProperties elasticsearchC
         return restClientBuilder(elasticsearchConfiguration).build();
     }
 
+    /**
+     * @param elasticsearchConfiguration The {@link DefaultElasticsearchConfigurationProperties} object
+     * @return The Elasticsearch Client
+     */
+    @Singleton
+    ElasticsearchClient elasticsearchClient(DefaultElasticsearchConfigurationProperties elasticsearchConfiguration, ObjectMapper objectMapper) {
```

Comment:
Needs `@Bean(preDestroy="shutdown")` since there is a shutdown method https://artifacts.elastic.co/javadoc/co/elastic/clients/elasticsearch-java/7.16.2/co/elastic/clients/elasticsearch/ElasticsearchClient.html#shutdown()

Reply:
That method returns an `ElasticShutdownClient` - name is a bit of a misnomer. If anything needs to be shut down, it's the underlying `ElasticsearchTransport` which would be a bit of a refactor.

Reply:
Exactly. That method returns an object that can shut down the cluster, not the client.

Reply:
In that case there should be a bean for the transport which is injected into `elasticsearchClient` and the async version and this bean should be shutdown

Reply:
did that